### DISCUSSION TITLE
Datatable and Paginator style/theme from PF 6.2 to 13.0

### DISCUSSION THREAD

**Discussion by xidbhonaker** [primefaces/community#214] [PrimeFaces] [answered]
I'm migrating an app from PrimeFaces 6.2 to 13.  It has some DataTable elements and I'm having difficulty getting the look to be similar.

The origianal paginator and header looked like this:

![image](https://github.com/primefaces/community/assets/22203855/b807b80c-4146-4296-8712-68c2064ae672)

The page controls look like buttons, all have a gradient, and the Filter for the first column is small.  Also, the grid itself has borders for each cell and the colors of each row alternate (there are variable heights).  And for expanding a row, I have an Icon:

![image](https://github.com/primefaces/community/assets/22203855/d919f8f5-b358-4735-b36c-a1f386f5dcb1)

All the functionality is there in 13.0.  But the visuals are different.  Here is the pagainator and header:

![image](https://github.com/primefaces/community/assets/22203855/7045be64-576b-48c8-8808-7e572eb7bc70)

And here is the girid.  While there are borders, they are very faint; and the backgrounds aren't alternating.  Plus the exapnd button is huge and not pretty.

![image](https://github.com/primefaces/community/assets/22203855/6b545611-0837-4a42-8230-d9af2bfe5012)

Some of this was accomplished with some minor CSS, but not very much.  I haven't dealt with Styles as I think that's a bigger change to the app.     Looking at the rendered HTML, there are CSS references in the downloaded theme that include gradient() and rgba() elements in the background. for example.  Also for some, there is a background-image reference to a URL on the website  url("/sitename/servlet/Scope/javax.faces.resource/images/ui-icons_38667f_256x240.png.xhtml?ln=primefaces-aristo").

Can someone point me in the right direction for documentation that can help?  Or better yet, point me to some working examples?

**Comment by NicolaIsotta** [comment]
Some of the things you want can be enabled with [p:dataTable attributes](https://primefaces.github.io/primefaces/13_0_0/#/components/datatable?id=attributes):

- `size` | regular | String | Size of the table content, valid values are "small" and "large". Leave empty for regular size.
- `stripedRows` | false | Boolean | Whether to display striped rows to visually separate content.
- `showGridlines` | false | Boolean | When enabled, cell borders are displayed.

**Reply by xidbhonaker** [reply]
Thanks, Nicolalsotta.  The size is the same as on 6.3, no issue there.  I already had showGridlines="true" in the screenshot above.  I added stripedRows="true", and here is a small screenshot.  

![image](https://github.com/primefaces/community/assets/22203855/440f722f-0de5-481e-834c-7d1dc0e04c72)

There are gridlines and alternating rows.  However they are barely visible.  I think this is because the active style for the grid includes "border: 1px solid rgba(0,0,0,.08);"... in other words, a black border with very high transparency.  

But the cells themselves now seem to a style of either tr.ui-widget-content.ui-datatable-odd or tr.ui-widget-content.ui-datatable-even.  They are exactly identical:

background: #fff;
transition: background-color .2s,color .2s,border-color .2s,box-shadow .2s,opacity .2s;

I really could use a good guide on how to the latest themes work, in order to fix this.  I seem to remember the old version had a page showcasing components (like datatable) that allowed you to switch between supported themes.  Is that still avaialable somewher?

**Comment by wnibes** [comment, marked as answer]
I'm doing exact the same thing (migrate PF 6.x -> 13) and it seems that there are many people who try to renew their frontend apps.

1) My intention is not have the same look. I would like to benefit from the advantages of the new skins. Responsiveness in PF 6.x was more difficult to achieve. That's why I only use the newer skins (they also look much better for my taste; our POC test in-house confirmed this with the overwhelming majority).
2) Sorting and Filtering of table data differs a bit (especially FilterMeta takes some attention).

To be more precise to your question: PF13 bundles other themes. "aristo" is not longer part of this. Maybe there is a solution to extract the aristo-theme-parts from the old PF6 library? But I, for myself, think that is not a good variant for the future...

**Comment by wnibes** [comment]
Some additional words on new themes:
If you manipulate margin and/or padding values, you can achieve a modified look on one theme, BUT the different themes come with different values for margins, paddings, insets, text sizes, etc., etc.
If these values are adjusted manually (CSS), this can lead to very unattractive effects when the theme is changed (e.g. the slightly larger text of one theme no longer fits into a field with a new, overwritten, absolute size of another theme).
That's why I'll try not to make any more "too hard" changes in future, so that I can switch between the themes without hesitation. In addition to the included themes, you can also consult commercial variants; perhaps there is something very compact that resembles your design...

**Reply by xidbhonaker** [reply]
Thanks, wnibes.  I intend to do just that in a future project,  Right now, though, I'm updating the app and environment to be newer Java and JSP and MyFaces... so am mostly just trying to keep the design as is at this point. (Volunteer work on a not-for-profit company website).   Those changes are going to include major design change too.  One thing at a time!

Can you point me to a site showing the 'existing' basic themes for 13.0, preferably with some components (like datatable) in the content?  I can't seem to find any on the main PrimeFaces documentation site.

**Reply by wnibes** [reply]
Sorry, I do not know such info.
Maybe the PF developers can help you with that.
But I would assume that the change of themes was not only made for visual reasons, but that the old themes are also no longer consistently compatible with the new components.
I also went through a long decision-making process and a few POC studies before coming to the conclusion that we should completely rebuild the applications. This is of course an enormous step, but in our case it also leads to the modernisation of the entire build stack (Java 17, Jakarta 10, PF13, etc.). For us, this is the better way, because we also really wanted a more modern, fresher design. How that fits for you is of course another question...
